Client/src/Components/Inputs/Autocomplete/index.jsx

```diff
@@ -0,0 +1,104 @@
+import { Autocomplete, Stack, TextField, Typography } from "@mui/material";
+import "./index.css";
+import React from "react";
+import PropTypes from "prop-types";
+import parse from "autosuggest-highlight/parse";
+import match from "autosuggest-highlight/match";
+import { useTheme } from "@emotion/react";
+
+/**
+ * AutoCompleteField component.
+ *
+ * @component
+ * @param {Object} props - The component props.
+ * @param {string} props.id - The ID of the autocomplete field.
+ * @param {string} props.type - The type of the input field (text or number).
+ * @param {Array} props.options - The options for the autocomplete field.
+ * @param {string} props.placeholder - The placeholder text for the input field.
+ * @param {boolean} props.disabled - Indicates if the field is disabled.
+ * @returns {JSX.Element} The AutoCompleteField component.
+ */
+
+const AutoCompleteField = ({
+  id,
+  type,
+  options,
+  placeholder = "Search",
+  disabled,
+}) => {
+  const [value, setValue] = React.useState();
+  const [inputValue, setInputValue] = React.useState("");
+  const theme = useTheme();
+
+  return (
+    <Autocomplete
+      className="auto-complete-field"
+      id={id}
+      value={value}
+      onChange={(event, newValue) => {
+        setValue(newValue);
+      }}
+      inputValue={inputValue}
+      onInputChange={(event, newInputValue) => {
+        setInputValue(newInputValue);
+      }}
+      options={options}
+      getOptionLabel={(option) => option.name}
+      renderInput={(params) => (
+        <div ref={params.InputProps.ref}>
+          <TextField
+            type={type}
+            {...params.inputProps}
+            placeholder={placeholder}
+            disabled={disabled}
+          />
+        </div>
+      )}
+      renderOption={(props, option, { inputValue }) => {
+        const { key, ...optionProps } = props;
+        const matches = match(option.name, inputValue, {
+          insideWords: true,
+        });
+        const parts = parse(option.name, matches);
+
+        return (
+          <li key={key} {...optionProps}>
+            <div>
+              {parts.map((part, index) => (
+                <span key={index}>{part.text}</span>
```

Comment:
Index [shouldn't be used as a key](https://react.dev/learn/rendering-lists#rules-of-keys) when rendering lists:

>You might be tempted to use an item’s index in the array as its key. In fact, that’s what React will use if you don’t specify a key at all. But the order in which you render items will change over time if an item is inserted, deleted, or if the array gets reordered. Index as a key often leads to subtle and confusing bugs.

Reply:
Fixed ✅